File: plugins/rhttp/android/src/main/java/com/flutter_rust_bridge/rhttp/RhttpPlugin.java
package com.flutter_rust_bridge.rhttp;

import android.content.Context;

import io.flutter.embedding.engine.plugins.FlutterPlugin;

public final class RhttpPlugin implements FlutterPlugin {
    static {
        System.loadLibrary("rhttp");
    }

    private static native boolean nativeInitPlatformVerifier(Context context);

    @Override
    public void onAttachedToEngine(FlutterPluginBinding binding) {
        if (!nativeInitPlatformVerifier(binding.getApplicationContext())) {
            throw new IllegalStateException(
                "Failed to initialize rustls-platform-verifier for Android"
            );
        }
    }

    @Override
    public void onDetachedFromEngine(FlutterPluginBinding binding) {}
}

File: plugins/rhttp/rhttp/android/src/main/kotlin/com/flutter_rust_bridge/rhttp/RhttpPlugin.kt
package com.flutter_rust_bridge.rhttp

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** RhttpPlugin */
class RhttpPlugin : FlutterPlugin, MethodCallHandler {
    companion object {
        init {
            System.loadLibrary("rhttp")
        }
    }

    private external fun initAndroid(ctx: Context)

    private external fun deinitAndroid()

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        initAndroid(flutterPluginBinding.applicationContext)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        deinitAndroid()
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        throw NotImplementedError()
    }
}
